### PR TITLE
network: several follow-ups for captive portal support

### PR DESCRIPTION
**Pull request by yuwata**
Follow-ups for #28132.

### CODE REVIEW

**Comment by yuwata** [comment]
cc @rpigott

**Review by rpigott — APPROVED** [review]
I don't really have a good setup for testing the ndisc stuff, but everything else LGTM and works.

**Comment by yuwata** [comment]
> I don't really have a good setup for testing the ndisc stuff,

Unfortunately, me neither, and dnsmasq seems not to support sending the portal through RA...

Anyway, thank you for the review.

File: src/network/networkd-state-file.c [hunk 1]
```diff
@@ -607,37 +608,12 @@ int link_save(Link *link) {
...
+                        return r;
 
-                if (dhcp_captive_portal)
-                        fprintf(f, "CAPTIVE_PORTAL=%s\n", dhcp_captive_portal);
-                else if (dhcp6_captive_portal)
-                        fprintf(f, "CAPTIVE_PORTAL=%s\n", dhcp6_captive_portal);
-                else if (link->ndisc_captive_portal)
-                        fprintf(f, "CAPTIVE_PORTAL=%s\n", link->ndisc_captive_portal);
+                if (captive_portal)
+                        fprintf(f, "CAPTIVE_PORTAL=%s\n", captive_portal);
```
Comment (poettering) [inline src/network/networkd-state-file.c:616]: hmm, this should be escaped, no? in case it contains a newline?

**Pull request merged**